File: Leetcode/sort_and_search/704_BinarySearch/Solution.java
class Solution {
    public int search(int[] nums, int target) {
        int start = 0, end = nums.length-1, mid;
        do{
            mid = start + (end - start)/2;
            if(nums[mid]==target) return mid;
            if(nums[mid]>target){
                end = mid-1;
            } else{
                start = mid+1;
            }
        }while(start<=end);
        return -1;
    }
}

File: Leetcode/sort_and_search/704_BinarySearch/Solution.py
class Solution(object):
    def search(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: int
        """
        if nums == []:  return -1
        start, end = 0, len(nums)-1
        while start<=end:
            mid = (end+start)//2
            if(nums[mid]==target):
                return mid
            elif nums[mid]>target:
                end = mid-1
            else:
                start = mid+1
        return -1
